src/systems.rs

```diff
@@ -221,17 +221,15 @@ pub fn movement(
                 if angle > std::f32::consts::PI {
                     angle -= 2.0 * std::f32::consts::PI;
                 }
-                (axis, angle)
+                axis * angle
             } else {
                 let current = tf.up();
-                (
-                    current.cross(settings.up_vector).normalize_or_zero(),
-                    current.angle_between(settings.up_vector),
-                )
+                current.normalize().cross(settings.up_vector.normalize())
```

Comment:
are you certain these shouldn't be `normalize_or_zero`?

Reply:
tbh I think I prefer panicking here if either up vector is zero, given that the user input and the transform up vector *should* be normalized already.